sql/src/main/java/io/crate/analyze/SQLPrinter.java

```diff
@@ -186,9 +192,22 @@ private void addOutput(StringBuilder sb, Field field, Symbol output) {
             }
         }
 
-        private static void addFrom(StringBuilder sb, QueriedRelation relation) {
+        private void addFrom(StringBuilder sb, QueriedTableRelation<?> relation) {
             sb.append(" FROM ");
-            sb.append(relation.getQualifiedName());
+            AbstractTableRelation<?> tableRelation = relation.tableRelation();
+            if (tableRelation instanceof TableFunctionRelation) {
+                QualifiedName qName = tableRelation.getQualifiedName();
+                Function function = ((TableFunctionRelation) tableRelation).function();
+                if (qName.getParts().size() == 2 && qName.getParts().get(1).equals(function.info().ident().name())) {
+                    sb.append(printSymbol(function));
+                } else {
+                    sb.append(printSymbol(function));
+                    sb.append(" AS ");
+                    sb.append(qName.toString());
+                }
+            } else {
+                sb.append(tableRelation.tableInfo().ident());
```

Comment:
Shouldn't it be `tableRelation.tableInfo().ident().sqlFqn()` so that the identifier is quoted if necessary?